### PR TITLE
Issue 1284

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
issue link: #1284 
## Check List
- check erc721 udts api holder count  correct
- check erc721 holders api correct
#### Test
- unit test
#### Task
 - none

### CODE REVIEW

File: lib/godwoken_explorer/graphql/resolovers/udt.ex [hunk 1]
```diff
@@ -460,21 +460,28 @@ defmodule GodwokenExplorer.Graphql.Resolvers.UDT do
...
+      |> distinct([cu], cu.address_hash)
+
     squery =
-      from(
-        cu in CurrentUDTBalance,
-        where: cu.value > 0
-      )
-      |> group_by([cu], cu.token_contract_address_hash)
-      |> select([cu], %{
+      from(a in Account, right_join: s in subquery(s0), on: a.eth_address == s.address_hash)
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/graphql/resolovers/udt.ex:472]: 这里没必要join account 吧

File: lib/godwoken_explorer/graphql/resolovers/udt.ex [hunk 2]
```diff
@@ -541,47 +548,42 @@ defmodule GodwokenExplorer.Graphql.Resolvers.UDT do
   def erc721_holders(_parent, %{input: input} = _args, _resolution) do
     contract_address = Map.get(input, :contract_address)
 
-    squery =
+    sq1 =
       from(cu in CurrentUDTBalance)
       |> where(
         [cu],
         cu.token_contract_address_hash == ^contract_address and cu.token_type == :erc721 and
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/graphql/resolovers/udt.ex:555]: erc721 在 CurrentUDTBalance 里是不知道 token_id 的吧？不是需要新建一个表去维护这个关系

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_explorer/graphql/resolovers/udt.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_explorer/graphql/resolovers/udt.ex]: fixed, just use subquery, clear `join account` code

File: lib/godwoken_explorer/graphql/resolovers/udt.ex (hunk 2 above)
Comment (Naupio) [reply lib/godwoken_explorer/graphql/resolovers/udt.ex:555]: holders  只需要显示 quantity, 可以通过现有逻辑计算出, token id 具体的 owner 不影响这个 api, 不过另一个表维护在另一个 pr 里面单独处理

**Comment by FrederLu** [comment]
https://godwoken-explorer-ui-development.vercel.app/nft-collection/0x78c88768b72db425836ae8abe93d35593f0d680c?tab=holders
image
image
Compared with the interface in the check list, it is found that there is a problem with the data. But it looks normal from the page.

**Review by FrederLu — APPROVED** [review]

**Review by zmcNotafraid — APPROVED** [review]

**Pull request merged**